From 48789f354ffe99aa28c08f55240e0b0a4deaa377 Mon Sep 17 00:00:00 2001
From: BogDan Vatra <bogdan@kdab.com>
Date: Mon, 1 Oct 2018 15:46:46 +0300
Subject: [PATCH] Android: fix compile with NDK r18+

In NDK r18, libc++.so was renamed to libc++.so.XX where XX is the Android
API level.

[ChangeLog][Android] Fixed build issue with NDK r18+.

Task-number: QTBUG-70631
Task-number: QTBUG-70779
Change-Id: Id0d2955648197e3054e3c69263b5a90d57964f6c
---
 mkspecs/android-clang/qmake.conf | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git x/qtbase/mkspecs/android-clang/qmake.conf y/qtbase/mkspecs/android-clang/qmake.conf
index b665000d00..1f5e690329 100644
--- x/qtbase/mkspecs/android-clang/qmake.conf
+++ y/qtbase/mkspecs/android-clang/qmake.conf
@@ -40,7 +40,11 @@ QMAKE_CFLAGS += -DANDROID_HAS_WSTRING --sysroot=$$NDK_ROOT/sysroot \
 ANDROID_SOURCES_CXX_STL_LIBDIR = $$NDK_ROOT/sources/cxx-stl/llvm-libc++/libs/$$ANDROID_TARGET_ARCH
 
 ANDROID_STDCPP_PATH = $$ANDROID_SOURCES_CXX_STL_LIBDIR/libc++_shared.so
-ANDROID_CXX_STL_LIBS = -lc++
+
+exists($$ANDROID_SOURCES_CXX_STL_LIBDIR/libc++.so): \
+    ANDROID_CXX_STL_LIBS = -lc++
+else: \
+    ANDROID_CXX_STL_LIBS = $$ANDROID_SOURCES_CXX_STL_LIBDIR/libc++.so.$$replace(ANDROID_PLATFORM, "android-", "")
 
 QMAKE_CFLAGS_OPTIMIZE_SIZE = -Oz
 
-- 
2.19.0

